test-helpers/src/contract_interaction/transfer_auth/mod.rs

```diff
@@ -33,56 +30,60 @@ pub fn gen_transfer_with_auth(
     permit2_address: Address,
     darkpool_address: Address,
     chain_id: u64,
-    account_addr: BigUint,
-    mint: BigUint,
-    amount: Amount,
-    is_withdrawal: bool,
+    transfer: ExternalTransfer,
+) -> Result<ExternalTransferWithAuth> {
+    match transfer.direction {
+        ExternalTransferDirection::Deposit => {
+            gen_deposit_with_auth(wallet, transfer, permit2_address, darkpool_address, chain_id)
+        },
+        ExternalTransferDirection::Withdrawal => gen_withdrawal_with_auth(wallet, transfer),
+    }
+}
+
+/// Generate a withdrawal payload with proper auth data
+fn gen_withdrawal_with_auth(
+    wallet: &Wallet<SigningKey>,
+    transfer: ExternalTransfer,
+) -> Result<ExternalTransferWithAuth> {
+    let transfer_bytes = serialize_calldata(&transfer)?;
```

Comment:
this will not form the correct signature. notice how before, we would convert to a `ContractExternalTransfer` via `to_contract_external_transfer`. we need to do this in the withdrawal case as it is the serialization of that struct that the contracts expect a signature over.